### ISSUE TITLE
Call to _SuspendThread inside native/proc.(*Process).stop sometimes fails with "Access is denied"

### ISSUE THREAD

**Issue by lloiser** [go-delve/delve#1170]
1. What version of Delve are you using (`dlv version`)?

```
Delve Debugger
Version: 1.0.0
Build: $Id: c98a142125d0b17bb11ec0513bde346229b5f533 $
```

2. What version of Go are you using? (`go version`)?

```
go version go1.10 windows/amd64
```

3. What operating system and processor architecture are you using?

Windows 10 Pro (10.0.16299 Build 16299)
x64 based architecture

4. What did you do?

Given the following code
```go
package main

import (
	"fmt"
	"log"
	"net/http"
	"time"
)

func main() {
	go func() {
		for {
			time.Sleep(time.Second * 3)
			fmt.Printf(".")
		}
	}()

	log.Fatal(http.ListenAndServe(":2080", nil))
}
```

When I start dlv headless (which f.e. happens in editors)
`dlv debug --headless=true --listen=localhost:2345 --api-version=2 --log=true`

And in another terminal connect to it:
`dlv connect :2345`

then run the following commands:
```
(dlv) c

===> ctrl-c here
received SIGINT, stopping process (will not forward signal)
Stopped at: 0x7ff810583801
=>no source available

(dlv) goroutine 1
Switched from 0 to 1 (thread 14568)

(dlv) frame 0
Stopped at: 0x7ff810583801
=>no source available
Frame 0: D:/Programs/Go/src/runtime/proc.go:292 (PC: 431b00)
   287:         mp.waittraceev = traceEv
   288:         mp.waittraceskip = traceskip
   289:         releasem(mp)
   290:         // can't do anything that might move the G between Ms here.
   291:         mcall(park_m)
=> 292: }
   293:
   294: // Puts the current goroutine into a waiting state and unlocks the lock.
   295: // The goroutine can be made runnable again by calling goready(gp).
   296: func goparkunlock(lock *mutex, reason string, traceEv byte, traceskip int) {
   297:         gopark(parkunlock_c, unsafe.Pointer(lock), reason, traceEv, traceskip)

(dlv) n
Command failed: Access is denied.

(dlv) n
Command failed: next while nexting
```

After this point I'm no longer able to exit or ctrl-c the `--headless` or the `connect` dlv process. I have to kill them manually using something else (e.g. windows taskmanager)

5. What did you expect to see?

Something similar as when I run `dlv debug --log`.

```
2018/04/02 21:59:33 server.go:73: Using API v1
2018/04/02 21:59:33 debugger.go:98: launching process with args: [D:\Development\src\github.com\lloiser\go-tests\fixtures\endless-bug\debug] Type 'help' for list of commands.

(dlv) c
2018/04/02 21:59:35 debugger.go:497: continuing
..
===> ctrl-c here
received SIGINT, stopping process (will not forward signal)
2018/04/02 21:59:42 debugger.go:486: halting
Stopped at: 0x7ff810583801
=>no source available

(dlv) goroutine 1
2018/04/02 21:59:45 debugger.go:525: switching to goroutine 1
Switched from 0 to 1 (thread 20392)

(dlv) frame 0
Stopped at: 0x7ff810583801
=>no source available
Frame 0: D:/Programs/Go/src/runtime/proc.go:292 (PC: 431b00)
   287:         mp.waittraceev = traceEv
   288:         mp.waittraceskip = traceskip
   289:         releasem(mp)
   290:         // can't do anything that might move the G between Ms here.
   291:         mcall(park_m)
=> 292: }
   293:
   294: // Puts the current goroutine into a waiting state and unlocks the lock.
   295: // The goroutine can be made runnable again by calling goready(gp).
   296: func goparkunlock(lock *mutex, reason string, traceEv byte, traceskip int) {
   297:         gopark(parkunlock_c, unsafe.Pointer(lock), reason, traceEv, traceskip)

(dlv) n
2018/04/02 21:59:58 debugger.go:509: nexting
Process has exited with status 2
```

6. What did you see instead?

The "Access is denied" error. And I was not able to exit dlv anymore...

**Comment by aarzilli** [comment]
I've investigated this. The problem:

1. doesn't have anything to do with headless mode (as far as I can tell)
2. is not deterministic

the way to reproduce this is to follow the instructions in the issue repeatedly until it happens. Eventually the debugger will try to stop at one of the conditional breakpoints set by `next` (to check the condition) but fail to stop one of the threads because of an "Access is Denied" error.
No clue what's causing this.

**Comment by bheimbaugh** [comment]
I can consistently recreate this (or something very similar) on 
```
C:\Windows\System32>go version
go version go1.10 windows/amd64

C:\Windows\System32>dlv version
Delve Debugger
Version: 1.0.0
Build: $Id: c98a142125d0b17bb11ec0513bde346229b5f533 $
```
See https://github.com/derekparker/delve/issues/556#issuecomment-393224075

**Comment by derekparker** [comment]
Closing as native OSX backend is not supported.

**Comment by aarzilli** [comment]
This is the windows backend though.

**Comment by aarzilli** [comment]
Closed by 0ae77bd6c5c7bffe5ae85447c790ea7d920fb593
